collect_app/src/androidTest/java/org/odk/collect/android/support/CollectHelpers.kt

```diff
@@ -57,4 +60,20 @@ object CollectHelpers {
                 it.save(ProjectKeys.KEY_SELECTED_GOOGLE_ACCOUNT, accountName)
             }
     }
+
+    fun killAndReopenApp() {
+        val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+
+        // kill
+        device.pressRecentApps()
```

Comment:
Wow I'd not realised you could do this! I think I tried something similar, but kept ending up with the test exiting as soon the app was killed.

A couple of things here:

1. I think that `killAndReopenApp` could just be a method on `Page`. To be really purist about it, it could take both a package name to restart and a destination page to assert on (here you'd pass `"org.odk.collect.android"` and a `MainMenuPage` instance).
2. It looks like this doesn't actual restart the process (`Application#onCreate` is not called again). We do already have a helper for simulating that (`CollectHelpers.simulateProcessRestart` could be called after the `swipeUp` though.

Reply:
> I think that killAndReopenApp could just be a method on Page. To be really purist about it, it could take both a package name to restart and a destination page to assert on (here you'd pass "org.odk.collect.android" and a MainMenuPage instance).

Done but do the package name I used `BuildConfig.APPLICATION_ID` so that we don't have to hardcode it.

> It looks like this doesn't actual restart the process (Application#onCreate is not called again). We do already have a helper for simulating that (CollectHelpers.simulateProcessRestart could be called after the swipeUp though.

Maybe the process is not restarted to keep the test running... never mind I've added `CollectHelpers.simulateProcessRestart()`.